syntax = "proto3";

package grpc_example;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

service Math {
  rpc Max (stream MaxRequest) returns (stream MaxResponse);
}

message MaxRequest {
  int32 a = 1;
  int32 b = 2;
}

message MaxResponse {
  int32 result = 1;
}